sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -72,6 +72,12 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   def requiredChildDistribution: Seq[Distribution] =
     Seq.fill(children.size)(UnspecifiedDistribution)
 
+  /** Specifies how data is ordered in each partition. */
+  def outputOrdering: Option[Ordering[Row]] = None
```

Comment:
I changed this from `Seq[SortOrder]` because you asked me not to manually bindReference in `SortMergeJoin.scala` but do it when constructing the RowOrdering. So I just create RowOrdering directly there.